--- include/chimera/key.h	2006-06-08 02:14:04.000000000 +0200
+++ include/chimera/key.h	2010-12-09 22:40:57.000000000 +0100
@@ -37,9 +37,8 @@
 
 
 /* global variables!! that are set in key_init function */
-Key Key_Max;
-Key Key_Half;
-
+/* stripped, not needed in the headers and would generate
+ * symbols in every binary including this header */
 
 /* key_makehash: hashed, s
 ** assign sha1 hash of the string #s# to #hashed# */
