--- src/route.c
+++ src/route.c
@@ -585,12 +585,13 @@ void leafset_update (ChimeraState * state, ChimeraHost * host, int joined,
 	}
     else
 	{
-	    if (key_between
-		(state->log, &host->key, &routeglob->me->key, &midpoint))
-		{
-		    leafset_delete (state, host, 1, deleted);
-		}
-	    else
+// Wenn wenige Knoten im Netzwerk sind, so kommen diese mehrmals im Leafset vor.
+// Daher loeschen wir sie links _und_ rechts heraus.
+// Johannes Held <uni@hehejo.de> 19.11.2010
+//	    if (key_between
+//		(state->log, &host->key, &routeglob->me->key, &midpoint))
+
+                leafset_delete (state, host, 1, deleted);
 		leafset_delete (state, host, 0, deleted);
 	}
 
